master/internal/proxy/proxy.go

```diff
@@ -174,14 +148,20 @@ func (p *Proxy) getSummary() map[string]Service {
 	return snapshot
 }
 
-func asyncCopy(dst io.Writer, src io.Reader) chan error {
+func asyncRun(f func() error) chan error {
```

Comment:
It feels like if we are writing an `asyncRun()` function in golang, then something is wrong.  In this case, I think what we should do is create a tiny little wrapper struct around a websocket that implements [io.Reader](https://golang.org/pkg/io/#Reader) and [io.Writer](https://golang.org/pkg/io/#Writer), and we can continue to use asyncCopy as before.